katz/src/main/kotlin/katz/NonEmptyList.kt

```diff
@@ -66,6 +66,6 @@ class NonEmptyList<out A> private constructor(
     }
 
     companion object Factory {
-        fun <A> of(head: A, vararg t: A): NonEmptyList<A> = NonEmptyList(head, t.asList())
+        fun <A: Any> of(head: A, vararg t: A): NonEmptyList<A> = NonEmptyList(head, t.asList())
     }
```

Comment:
I am not sure about the type bound in of(). Since NonEmptyList is already parameterized with <A: Any> it is probably redundant on the companion object function. It may make the code more clear but it can also be seen as a duplicate code / specification. What do you think?